### PR TITLE
soc: add STM32F072X8 SOC

### PR DESCRIPTION
**Pull request by ottojo**
basically just copied from the existing `stm32f072Xb`, from which it only differs in having less flash

I use this to run zephyr on the [otterpill](https://github.com/Jana-Marie/OtterPill) board, which i plan to upstream (from [here](https://github.com/ottojo/zephyr_sandbox/tree/main/oot_module/boards/arm/otterpill)) as well. if thats more convenient, i can also combine the PRs once the board-stuff is ready...

### CODE REVIEW

File: soc/arm/st_stm32/stm32f0/Kconfig.soc [hunk 1]
```diff
@@ -34,6 +34,9 @@ config SOC_STM32F070XB
 config SOC_STM32F072XB
 	bool "STM32F072XB"
 
+config SOC_STM32F072X8
```
Comment (gautierg-st) [inline soc/arm/st_stm32/stm32f0/Kconfig.soc:37]: Nitpick: Add that just before F072XB, to keep them in "alphabetical" order

**Review by gautierg-st — CHANGES_REQUESTED** [review]

File: soc/arm/st_stm32/stm32f0/Kconfig.defconfig.stm32f072x8 [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# ST Microelectronics STM32F072X8 MCU
+
+if SOC_STM32F072X8
+
+# STM HAL uses 'stm32f072xb' for both STM32F072XB and STM32F072X8 variants.
```
Comment (erwango) [inline soc/arm/st_stm32/stm32f0/Kconfig.defconfig.stm32f072x8:5]: In that case, please change `Kconfig.defconfig.stm32f072xb` file name to `Kconfig.defconfig.stm32f072xx` and factorize definitions.
Check [here](https://github.com/zephyrproject-rtos/zephyr/blob/main/soc/arm/st_stm32/stm32f1/Kconfig.defconfig.stm32f105xx) for an example.

File: dts/arm/st/f0/stm32f072X8.dtsi [hunk 3]
```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright (c) 2018 Linaro Limited
```
Comment (erwango) [inline dts/arm/st/f0/stm32f072X8.dtsi:2]: Please use your own Copyright.
Even if this is a copy/paste you're the one doing the upstream and (hopefully) validation effort.

**Review by erwango — CHANGES_REQUESTED** [review]
Thanks for bringing in this change. Some comments

File: soc/arm/st_stm32/stm32f0/Kconfig.soc (hunk 1 above)
Comment (ottojo) [reply soc/arm/st_stm32/stm32f0/Kconfig.soc:34]: fixed it!

File: soc/arm/st_stm32/stm32f0/Kconfig.defconfig.stm32f072x8 (hunk 2 above)
Comment (ottojo) [reply soc/arm/st_stm32/stm32f0/Kconfig.defconfig.stm32f072x8]: i combined both files into Kconfig.defconfig.stm32f072xx, as in the example :+1:

File: dts/arm/st/f0/stm32f072X8.dtsi (hunk 3 above)
Comment (ottojo) [reply dts/arm/st/f0/stm32f072X8.dtsi]: done!

**Review by erwango — APPROVED** [review]

**Review by gautierg-st — APPROVED** [review]

**Pull request merged**